designer/server/src/main/scala/pl/touk/nussknacker/ui/factory/NussknackerAppFactory.scala

```diff
@@ -42,49 +108,252 @@ class NussknackerAppFactory(
+        deploymentsStatusesProvider,
+        actionRepository,
+        dbioRunner
+      )(executionContextWithIORuntime)
+      _ <- FinishedDeploymentsStatusesSynchronizationScheduler.resource(
+        system,
+        reconciler,
+        FinishedDeploymentsStatusesSynchronizationConfig.parse(resolvedDesignerConfig)
+      )
+
+      authenticationResources =
+        AuthenticationResources(resolvedDesignerConfig, getClass.getClassLoader, futureSttpBackend)(
+          executionContextWithIORuntime
+        )
+      authManager = new AuthManager(authenticationResources)(executionContextWithIORuntime)
+
+      _ = Initialization.init(
+        migrations,
+        dbRef,
+        clock,
+        processRepository,
+        scenarioActivityRepository,
+        scenarioLabelsRepository,
+        environment
+      )(executionContextWithIORuntime)
+
+      scenarioStatusPresenter = new ScenarioStatusPresenter(dmDispatcher)
+
+      fragmentRepository = new DefaultFragmentRepository(futureProcessRepository)(executionContextWithIORuntime)
+      fragmentResolver   = new FragmentResolver(fragmentRepository)
+
+      counter = new ProcessCounter(fragmentRepository)
+
+      processingTypeServicesProvider = processingTypeDataProvider.mapValues(
+        ProcessingTypeServices.create(
+          resolvedDesignerConfig,
+          featureTogglesConfig,
+          additionalUIConfigProvider,
+          fragmentRepository,
+          fragmentResolver,
+          counter,
+          _
+        )(executionContextWithIORuntime)
+      )
+
+      processService = new DBProcessService(
+        scenarioStatusProvider,
+        scenarioStatusPresenter,
+        processingTypeServicesProvider.mapValues(_.newProcessPreparer),
+        processingTypeDataProvider.mapCombined(_.parametersService),
+        processingTypeServicesProvider.mapValues(_.processResolver),
+        dbioRunner,
+        futureProcessRepository,
+        actionRepository,
+        writeProcessRepository,
+      )(executionContextWithIORuntime)
+
+      customHttpServiceProviders <- createCustomHttpServiceProvider(resolvedDesignerConfig, processService)(
+        executionContextWithIORuntime
+      )
+
+      fingerprintService = new FingerprintService(new FingerprintRepositoryImpl(dbRef)(executionContextWithIORuntime))(
+        executionContextWithIORuntime,
+        dbioRunner
       )
-      _ <- server.start(alreadyLoadedConfig, metricsRegistry)
+
+      componentService = {
+        new DefaultComponentService(
+          ComponentLinksConfigExtractor.extract(resolvedDesignerConfig),
+          processingTypeServicesProvider.mapValues(_.componentServiceProcessingTypeData),
+          processService,
+          fragmentRepository
+        )(executionContextWithIORuntime)
+      }
+      processAuthorizer = new AuthorizeProcess(futureProcessRepository)(executionContextWithIORuntime)
+      route = new AkkaHttpBasedRouteFactory(
```

Comment:
This method (the `createApp`) is getting big. I see there is no easy refactor, but it seems we could extract the route creation to a separate, private method. Some of the dependencies needed by the AkkaHttpBasedRouteFactory are used only in this context, so could be easily moved to this method (but at the same time I'm avare that the new method will have many params)

Reply:
I did it a little bit different, take a look